flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/delegation/StreamPlanner.scala

```diff
@@ -129,14 +134,48 @@ class StreamPlanner(
     new StreamPlanner(executor, config, moduleManager, functionCatalog, catalogManager)
   }
 
+  override def loadPlan(planReference: PlanReference): CompiledPlanInternal = {
+    val ctx = createSerdeContext
+    val objectReader: ObjectReader = JsonSerdeUtil.createObjectReader(ctx)
+    val execNodeGraph = planReference match {
+      case filePlanReference: FilePlanReference =>
+        objectReader.readValue(filePlanReference.getFile, classOf[ExecNodeGraph])
+      case contentPlanReference: ContentPlanReference =>
+        objectReader.readValue(contentPlanReference.getContent, classOf[ExecNodeGraph])
+      case resourcePlanReference: ResourcePlanReference => {
+        val url = resourcePlanReference.getClassLoader
+          .getResource(resourcePlanReference.getResourcePath)
+        if (url == null) {
+          throw new IOException(
+            "Cannot load the plan reference from classpath: " + planReference);
+        }
+        objectReader.readValue(new File(url.toURI), classOf[ExecNodeGraph])
+      }
+      case _ => throw new IllegalStateException(
+        "Unknown PlanReference. This is a bug, please contact the developers")
+    }
+
+    new ExecNodeGraphCompiledPlan(
+      this,
+      JsonSerdeUtil.createObjectWriter(createSerdeContext)
+        .withDefaultPrettyPrinter()
+        .writeValueAsString(execNodeGraph),
+      execNodeGraph)
+  }
+
   override def compilePlan(modifyOperations: util.List[ModifyOperation]): CompiledPlanInternal = {
     validateAndOverrideConfiguration()
     val relNodes = modifyOperations.map(translateToRel)
     val optimizedRelNodes = optimize(relNodes)
     val execGraph = translateToExecNodeGraph(optimizedRelNodes)
     cleanupInternalConfigurations()
 
-    new ExecNodeGraphCompiledPlan(createSerdeContext, execGraph)
+    new ExecNodeGraphCompiledPlan(
+      this,
+      JsonSerdeUtil.createObjectWriter(createSerdeContext)
+        .withDefaultPrettyPrinter()
```

Comment:
shall we only pretty print for the `CompiledPlan.printJsonString()`? I guess pretty printing has impact on the plan sizes.